README.md

```diff
@@ -1 +1,63 @@
-# angular-what-is-a-custom-directive-readme
\ No newline at end of file
+# What is a custom Directive?
+
+## Overview
+
+One of the massive powerful aspects of Angular is the ability to actually be able to create our own directives.
+
+## Objectives
+
+- Describe what components are
+- Describe custom Directives
+- Create a custom Directive
+
+## Components
+
+Nowadays, most frontend applications are made up of small, reusable components. If we imagine a car, it would be made up with components such as wheels, windows, engine, etc.
+
+We can create these components using directives. We've already used directives that Angular gives us, but we can also make our own.
+
+Directives can both be HTML attributes and HTML elements. Most of our directives will be HTML elements, as they will have their own template inside of themselves instead of modifying the behaviour of existing elements.
+
+If we imagine an `ng-repeat`, that modifies the behaviour of the element it is placed on by repeating it for every item in an array.
+
+However, we might create a custom directive called "dropdown". This dropdown directive will have it's own template that will get put into the DOM, so we create it as a HTML element instead.
+
+## Creating a custom directive
+
+Much like services and controllers, we can use the `.directive` function to create our directives.
+
+Our directive names are what we then use to reference them in the DOM. However, as the DOM is case-insensitive, we change the captial letters in our name to hypens. For instance, the directive name `myDirective` becomes `<my-directive></my-directive>`.
```

Comment:
Nice example on the directive names - add a second example here so that the user can solidify the pattern. `userDropdownMenu` becomes `<user-dropdown-menu></user-dropdown-menu>` or something.